File: 58.length-of-last-word.js
/*
 * @lc app=leetcode id=58 lang=javascript
 *
 * [58] Length of Last Word
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    //s = "   fly me   to   the moon  "

    let removedSpaces = s.replace(/\s+/g, " "); //removing whitespaces in between
    
    //removing trailing spaces and splitting string into array
    let stringArray = removedSpaces.trim().split(" ");

    return stringArray[stringArray.length-1].length;
};
// @lc code=end

File: 58.length-of-last-word.java
/*
 * @lc app=leetcode id=58 lang=java
 *
 * [58] Length of Last Word
 */

// @lc code=start
class Solution {
    public int lengthOfLastWord(String s) {
        
        int lastLength = 0;
        String[] wordArray = s.split(" ");

        for(int i = wordArray.length-1; i >= 0; i--){
            if(wordArray[i].length() > 0){
                lastLength = wordArray[i].length();
                break;
            }
        }

        return lastLength;
    }
}
// @lc code=end

